coldfront/core/project/views_/new_project_views/request_views.py

```diff
@@ -460,34 +509,50 @@ def __handle_pi_data(self, form_data):
         step_number = self.step_numbers_by_form_name['existing_pi']
         data = form_data[step_number]
         if data['PI']:
-            return data['PI']
-
-        # Create a new User object intended to be a new PI.
-        step_number = self.step_numbers_by_form_name['new_pi']
-        data = form_data[step_number]
-        try:
-            email = data['email']
-            pi = User.objects.create(
-                username=email,
-                first_name=data['first_name'],
-                last_name=data['last_name'],
-                email=email,
-                is_active=False)
-        except IntegrityError as e:
-            self.logger.error(f'User {email} unexpectedly exists.')
-            raise e
-
-        # Set the user's middle name in the UserProfile; generate a PI request.
-        try:
+            pi = data['PI']
             pi_profile = pi.userprofile
-        except UserProfile.DoesNotExist as e:
-            self.logger.error(
-                f'User {email} unexpectedly has no UserProfile.')
-            raise e
-        pi_profile.middle_name = data['middle_name']
-        pi_profile.upgrade_request = utc_now_offset_aware()
-        pi_profile.save()
+        else:
+            # Create a new User object intended to be a new PI.
+            step_number = self.step_numbers_by_form_name['new_pi']
+            data = form_data[step_number]
 
+            try:
+                email = data['email']
+                pi = User.objects.create(
+                    username=email,
+                    first_name=data['first_name'],
+                    last_name=data['last_name'],
+                    email=email,
+                    is_active=False)
+            except IntegrityError as e:
+                self.logger.error(f'User {email} unexpectedly exists.')
+                raise e
+
+            # Set user's middle name in the UserProfile; generate a PI request.
+            try:
+                pi_profile = pi.userprofile
+            except UserProfile.DoesNotExist as e:
+                self.logger.error(
+                    f'User {email} unexpectedly has no UserProfile.')
+                raise e
+            pi_profile.middle_name = data['middle_name']
+            pi_profile.upgrade_request = utc_now_offset_aware()
+            pi_profile.save()
+
+        if flag_enabled('USER_DEPARTMENTS_ENABLED'):
+            # Set the user's authoritative departments in the UserProfile.
+            
+            fetch_and_set_user_departments(pi, pi_profile)
```

Comment:
Would both authoritative and non-authoritative ever be set at the same time? Can we use an `if-else` to make that explicit?

Reply:
I don't think so. Done